### ISSUE TITLE
First timestamp of CUE sheet is bogus after restarting a recording

### ISSUE THREAD

**Issue by fwcd** [mixxxdj/mixxx#13964]
### Bug Description

To reproduce:

- Play a track
- Start a recording
- Let it record a moment
- Stop the recording
- Start a new recording
- Stop the new recording

In the first recording the first track starts at 00:00:00 (as expected), e.g.:

```
FILE "2024-12-03_00h28m23s.wav" WAVE
  TRACK 01 AUDIO
    TITLE "Harder, Better, Faster, Stronger"
    PERFORMER "Daft Punk"
    INDEX 01 00:00:00
```

In the second recording, however, the first timestamp is not 00:00:00 despite the track even though the track already played when pressing the record button:

```
FILE "2024-12-03_00h28m31s.wav" WAVE
  TRACK 01 AUDIO
    TITLE "Harder, Better, Faster, Stronger"
    PERFORMER "Daft Punk"
    INDEX 01 00:05:00
```

Further investigation showed that this timestamp (00:05:00) is actually the _length of the first recording_, which shouldn't have any effect on further recordings. All subsequent timestamps are fine, however.

This seems to be a really old bug, I can reproduce it all the way back to Mixxx 2.3.0 (haven't checked older versions).

### Version

2.6-alpha-138-g0eccb1b2db (HEAD)

### OS

macOS 15.1.1

**Comment by fwcd** [comment]
Digging through the code, this looks suspicious:

https://github.com/mixxxdj/mixxx/blob/9f43c3e798bb05f835ac9ee85403317586723bb1/src/engine/sidechain/enginerecord.cpp#L204-L215

`m_recordedDuration` has likely not been reset to 0 after starting a new recording. It seems like this logic was refactored way back in 2016 in https://github.com/mixxxdj/mixxx/commit/3e4902ca1543c5c907823368b75b68300015b2c8 (in #1035, which was included in 2.2.4).

**Comment by fwcd** [comment]
Interestingly, the issue was fixed in 3e4902ca1543c5c907823368b75b68300015b2c8 for split recordings:

https://github.com/mixxxdj/mixxx/blob/9f43c3e798bb05f835ac9ee85403317586723bb1/src/engine/sidechain/enginerecord.cpp#L178-L182

...but not for new recordings:

https://github.com/mixxxdj/mixxx/blob/9f43c3e798bb05f835ac9ee85403317586723bb1/src/engine/sidechain/enginerecord.cpp#L146-L149

So I assume this is an omission.
